### PR TITLE
eval: improve ex_execute

### PR DESCRIPTION
**Pull request by erw7**
Change to use `encode_tv2echo` instead of `tv_stringify`.

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -9309,16 +9288,18 @@ void ex_execute(exarg_T *eap)
     }
 
     if (!eap->skip) {
-      char buf[NUMBUFLEN];
       const char *const argstr = eap->cmdidx == CMD_execute
-        ? tv_get_string_buf(&rettv, buf)
-        : tv_stringify(&rettv, buf);
+        ? tv_get_string(&rettv)
```
Comment (teto) [inline src/nvim/eval.c:9292]: please keep the `tv_get_string_buf`, we want purer functions.

Comment (teto) [reply src/nvim/eval.c:9292]: was tired I guess, tv_get_string is ofc fine here.

**Comment by teto** [comment]
could you rebase to trigger CI please ?

**Comment by erw7** [comment]
> could you rebase to trigger CI please ?

Done.

**Pull request merged**